### DISCUSSION TITLE
did "find" keyboarding/shortcut change

### DISCUSSION THREAD

**Discussion by reagle** [freeplane/freeplane#666] [Help]
I noticed that with 1.10.4 find behavior is not as I expected. I do have it mapped to `cmd+g`, which, if I press, creates a popup in which I can enter text to search for. However, pressing enter does **NOT** find the first instance. It just dismisses the popup and my search term now appears in the search box in the menu bar. Then I have to press `cmd+shift+g` to find the first instance. I'm confident `enter` within the search popup took me to the first instance. What changed?

![SCR-20220914-flo](https://user-images.githubusercontent.com/596814/190194329-69e23541-01a0-42cc-8629-392dd4a6d8c3.png)

freeplane: 1.10.4 on macOS 12.5.1 21G83 arm64

**Comment by reagle** [comment]
Returning to 1.10.2 the problem goes away. (I didn't try 1.10.3.)

**Comment by dpolivaev** [comment]
Now you can click on the right and left arrows to the left of the input field.
I have noticed that currently they can not be selected using the keyboard, and I a going to fix it.

**Reply by reagle** [reply]
Yes, I don't want to have to use the mouse! 😄 `enter` worked pretty well.

**Reply by dpolivaev** [reply]
Yes, it is how it goes: some people want to have buttons and I implement buttons, and then some other people want to use the keyboard and I add some key listeners. I am going to use control + arrow left / right.

**Reply by dpolivaev** [reply]
Done

<img width="337" alt="find" src="https://user-images.githubusercontent.com/1558052/190263860-26dad154-3cd1-4418-9298-448c7bdebee1.png">

**Comment by reagle** [comment]
Thank you. Let me know which release it'll be in and I'll give it a try.

**Reply by dpolivaev** [reply]
Preview 1.10.5_03

**Reply by reagle** [reply]
I'm running 1.10.6 and am trying to use the keys I see on hover over the arrow, which are: ^+→ . I take that as control and right cursor/arrow. However, when I press that I get an error dialog "The root node can't be added to a new parent."

<img width="367" alt="SCR-20230110-ict" src="https://user-images.githubusercontent.com/596814/211630621-ec33b0d2-804f-426f-9ae3-aa9ed96cd4f7.png">

<img width="614" alt="SCR-20230110-icp" src="https://user-images.githubusercontent.com/596814/211630627-e2e94a48-b16a-4296-b2ee-161a99a98b5c.png">

**Comment by reagle** [comment]
Perhaps it's that I have to keep the find dialog for those keyboard shortcuts to work. In the past, I think I could continue to search for the next instance by keyboard, even when I had dismissed the search dialog...?

**Comment by reagle** [comment]
Yes, in the past, I could hit `cmd+g`, type in my query, hit `return`, the cursor would then go to the first instance, and I could hit other keys to proceed. Now, I type `cmd+g`, type in my query, do NOT his `return`, then hit `ctrl+arrow` to move the cursor/highlight to the relevant text with the find dialog obscuring my view.
